### PR TITLE
'MDAnalysis.analysis.nucleicacids' parallelization

### LINKED ISSUE #4670: `MDAnalysis.analysis.nucleicacids`: Implement parallelization or mark as unparallelizable
Given a newly implemented parallelization (https://docs.mdanalysis.org/dev/documentation_pages/analysis/parallelization.html), implement it for all classes in `analysis.nucleicacids` module. You may use `analysis.RMSD` in `develop` as a reference.

You need to add:

 - appropriate implementation of result aggregation for AnalysisBase subclasses in `analysis.nucleicacids` module (or explicitly limit available backends if the classes aren't parallelizable)
 - a boilerplate fixture(s) to `testsuite/analysis/conftest.py`, analogous with existing ones
 - a `client_...` fixtures to all tests using in `testsuite/MDAnalysisTests/analysis/test_nucleicacids.py`, and modify the way run() method is called

### PR DESCRIPTION
**Pull request by talagayev**
Fixes #4670 attempt

Changes made in this Pull Request:
 - added backends and aggregators to `NucPairDist` in `analysis.nucleicacids`
 - added the `client_NucPairDist` in `conftest.py`
 - added `client_NucPairDist` in `run()` in `test_nucleicacids.py`

Here the Error:

`AttributeError: 'WatsonCrickDist' object has no attribute '_res_array'` appears for all of the tests where the `client_NucPairDist` is used, same goes not only for `WatsonCrickDist`, but also for `MinorPairDist` and `MajorPairDist`.

I am a little bit stuck, since I am not sure if I need to modify the `analysis.nucleicacids` to fix this error, thus I made this PR as a draft to get more input and ideas how to fix this issue :)

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
📚 Documentation preview 📚: https://mdanalysis--4727.org.readthedocs.build/en/4727/

### CODE REVIEW

File: package/MDAnalysis/analysis/nucleicacids.py
Comment (yuxuanzhuang) [inline package/MDAnalysis/analysis/nucleicacids.py:177]: Not related to this PR, but I don't think `_res_dict` (the line below) is used anymore in the code.

File: package/MDAnalysis/analysis/nucleicacids.py
Comment (yuxuanzhuang) [inline package/MDAnalysis/analysis/nucleicacids.py:167]: add version changed

**Review by yuxuanzhuang — CHANGES_REQUESTED** [review]
Looking good! Could you also help fix the pep8?

**Comment by talagayev** [comment]
> Looking good! Could you also help fix the pep8?

Yes, I have now access again to a PC, although with slow internet, so I will adjust it for pep8 and also add the additional stuff that is missing (changelog etc.).

I would also create an additional PR draft for `analysis.align`, since there I am not sure if it can be parallelized :)

File: package/MDAnalysis/analysis/nucleicacids.py
Comment (talagayev) [reply package/MDAnalysis/analysis/nucleicacids.py:177]: Adjusted the PEP8 issues, should we keep `_res_dict` for now?

File: package/MDAnalysis/analysis/nucleicacids.py
Comment (talagayev) [reply package/MDAnalysis/analysis/nucleicacids.py:167]: Added the versionchange and also mentioned there the 'self._res_array' was modified to 'self.results.distances'.

File: package/MDAnalysis/analysis/nucleicacids.py
Comment (yuxuanzhuang) [reply package/MDAnalysis/analysis/nucleicacids.py:177]: You can just remove it; don't think theres's any need to add notes, as it’s only used internally and for type hinting.

**Review by yuxuanzhuang — APPROVED** [review]
lgtm!

File: package/MDAnalysis/analysis/nucleicacids.py
Comment (talagayev) [reply package/MDAnalysis/analysis/nucleicacids.py:177]: Done, true makes sense, since it is not used in the code after that.

Comment (talagayev) [reply package/MDAnalysis/analysis/nucleicacids.py:177]: okay removing `_res_dict` led to the pytests failing, I will bring it back again.

Comment (talagayev) [reply package/MDAnalysis/analysis/nucleicacids.py:177]: nah my bad, it was some other failure not connected to this code, so it should be fine

**Comment by orbeckst** [comment]
@yuxuanzhuang thanks for the review. Could you please shepherd the PR to completion?

(I think we're in feature-freeze for 2.8.0 #4733 but check the issue for updates.)

File: package/MDAnalysis/analysis/nucleicacids.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/nucleicacids.py]: Remove
```suggestion
```
The old `_res_array` was never public so it doesn't have to be mentioned in user-facing docs.

**Comment by talagayev** [comment]
@yuxuanzhuang for this one, since it is also approved it can be merged, but now with the new release the `CHANGELOG` also changes. should I modify it?

If yes would it be then `2.9.0 `

**Review by IAlibay — CHANGES_REQUESTED** [review]
Please don't modify the changelog @talagayev - I'm in the process of finalizing the release. Once I've merged in the changelog update for 2.9.0 then folks can merge more things.

**Comment by talagayev** [comment]
Hey @IAlibay, perfect sounds good 😸

**Review by IAlibay — APPROVED** [review]
Alright you're good to go @talagayev - just update against `develop` and it should be ok (I'm approving as "unblock" I didn't actually read the contents of this PR sorry).

**Comment by talagayev** [comment]
> Alright you're good to go @talagayev - just update against `develop` and it should be ok (I'm approving as "unblock" I didn't actually read the contents of this PR sorry).

Updated it, should be ready :) All good, was mainly the parallelization implementation similar to some other PR request that I did, with @yuxuanzhuang looking at them

**Comment by talagayev** [comment]
Ah need to move it to 2.9.0 in the `CHANGELOG`, will do it now

**Comment by orbeckst** [comment]
@yuxuanzhuang I think this one is good to go but please have a last look over.

File: package/MDAnalysis/analysis/nucleicacids.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/nucleicacids.py]:
```suggestion
    .. versionchanged:: 2.9.0
```

File: package/CHANGELOG
Comment (orbeckst) [inline package/CHANGELOG:106]: normally we don't change previous CHANGELOG entries but that's harmless enough

Comment (talagayev) [reply package/CHANGELOG:106]: Ah I see, thought that maybe there is a regulation for the Capitalization in the Changelogs, that I forgot to take
into account, so I changed the capitalization for the Entries, but will know for the future :)

**Review by RMeli — APPROVED** [review]

**Comment by yuxuanzhuang** [comment]
@talagayev Thanks for your contribution!

**Comment by talagayev** [comment]
> @talagayev Thanks for your contribution!

Happy to help :) and thanks to you too, for helping with the PR :)

**Pull request merged**
